tests/test_rich_markup_mode.py

```diff
@@ -38,3 +41,186 @@ def main(arg: str):
+        ("none", ["First line", "Line 1", "Line 2", "Line 3", ""]),
+    ],
+)
+def test_markup_mode_newline_pr815(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """First line
+
+        Line 1
+
+        Line 2
+
+        Line 3
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = result_lines.index("First line")
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        (
+            "markdown",
+            [
+                "This header is just pretty long really",
+                "Line 1 of a very extremely super long line And a short line 2",
+                "",
+            ],
+        ),
+        (
+            "rich",
+            [
+                "This header is just pretty long really",
+                "Line 1 of a very",
+                "extremely super long",
+                "line",
+                "",
+                "And a short line 2",
+                "",
+            ],
+        ),
+        (
+            "none",
+            [
+                "This header is just pretty long really",
+                "Line 1 of a very extremely super long line",
+                "And a short line 2",
+                "",
+            ],
+        ),
+    ],
+)
+def test_markup_mode_newline_issue447(mode: str, lines: List[str]):
+    app = typer.Typer(rich_markup_mode=mode)
+
+    @app.command()
+    def main(arg: str):
+        """This header
+        is just
+        pretty long
+        really
+
+        Line 1 of a very
+        extremely super long
+        line
+
+        And a short line 2
+        """
+        print(f"Hello {arg}")
+
+    assert app.rich_markup_mode == mode
+
+    result = runner.invoke(app, ["World"])
+    assert "Hello World" in result.stdout
+
+    result = runner.invoke(app, ["--help"])
+    result_lines = [line.strip() for line in result.stdout.split("\n")]
+    assert any(c in result.stdout for c in rounded)
+    help_start = [i for i, row in enumerate(result_lines) if "This header" in row][0]
+    arg_start = [i for i, row in enumerate(result_lines) if "Arguments" in row][0]
+    assert help_start != -1
+    assert result_lines[help_start:arg_start] == lines
+
+
+@pytest.mark.parametrize(
+    "mode,lines",
+    [
+        pytest.param(
+            "markdown",
+            ["First line", "", "• 1", "• 2", "• 3", ""],
+            marks=pytest.mark.xfail(),
+        ),
```

Comment:
This test currently fails on `master`, with the output being 
```
['First line', '', '• 1 - 2 - 3', '']
```
-> this happens mostly because the newlines are not parsed correctly, after which the bullet points aren't recognized correctly. PR #815 attempts to fix this.

Reply:
Hum, this gets very tricky. :thinking: 

Because I see how I would intuitively expect the lines to be split, visually it's just a list.

But then, that would go against the automatic text unwrapping that Click would do and that I was mentioning above. :thinking: 

One option is to expect it all to be unwrapped, so that if a user wants a list they would separate the items by new lines.

The other option is to challenge the first idea of automatically unwrapping new lines into a single paragraphs, and instead render things as-is, even if they have internal splits.

A third option I wouldn't want is adding a special case for lists, but that would open many edge cases and many other possible edge cases to handle, so I think I wouldn't like this idea, maybe better to decide on one format or the other.